src/classes/DragNDropController.cls

```diff
@@ -0,0 +1,156 @@
+     */
+    @AuraEnabled
+    public static ContentVersion getProfilePicture(Id parentId) {
+        ContentVersion contentVersion;
+
+        try {
+            List<ContentDocumentLink> cdl = [
+                SELECT Id, ContentDocumentId
+                FROM ContentDocumentLink
+                WHERE LinkedEntityId = :parentId
+                LIMIT 1
+                ];
+            
+            if (!cdl.isEmpty()) {
+                Id contentDocumentId = cdl[0].contentDocumentId;
+
+                contentVersion = [
+                    SELECT VersionData, IsLatest
+                    FROM ContentVersion
+                    WHERE ContentDocumentId = :contentDocumentId AND IsLatest = true
+                    ];
+            }
+        } catch (QueryException e) {
+            // Throw an AuraHandledException with custom data.
+            CustomExceptionData data = new CustomExceptionData(
+                'There was an error trying to get the Content Version',
+                e.getMessage()
+                );
+            throw new AuraHandledException(JSON.serialize(data));
+        }
+
+        return contentVersion;
+    }
+
+    /**
+     * Create Content Version record in order to upload the image and associate the record with the parentId.
+     *
+     * @param parentId The record ID.
+     * @param fileName The file name.
+     * @param base64Data The blob image.
+     * @param contentType The file type/extension.
+     */
+    @AuraEnabled
+    public static void saveFile(Id parentId, String fileName, String base64Data, String contentType) {
+        Id contentDocumentId;
+        Savepoint sp;
+
+        try {
+            List<ContentDocumentLink> cdl = [
+                SELECT Id, ContentDocumentId
+                FROM ContentDocumentLink
+                WHERE LinkedEntityId = :parentId
+                LIMIT 1
+                ];
+
+            if (!cdl.isEmpty()) {
+                contentDocumentId = cdl.get(0).contentDocumentId;
+            }
+
+            sp = Database.setSavepoint();
+
+            ContentVersion cv = new ContentVersion();
+            cv.ContentLocation = 'S';
+            cv.ContentDocumentId = contentDocumentId;
+            cv.VersionData = EncodingUtil.base64Decode(base64Data);
+            cv.Title = fileName;
+            cv.PathOnClient = filename;
+            insert cv;
+
+            if (contentDocumentId == null) {
+                ContentDocumentLink newLink = new ContentDocumentLink(LinkedEntityId = parentId, ShareType = 'V');
+
+                newLink.ContentDocumentId = [
+                    SELECT Id, ContentDocumentId
+                    FROM ContentVersion
+                    WHERE Id = :cv.Id
+                    ].ContentDocumentId;
+
+                insert newLink;
+            }
+        } catch (QueryException e) {
+            // Throw an AuraHandledException with custom data.
```

Comment:
We should also have a `Database.rollback(sp);` here since it could fail in the second soql query after te content version insertion.